src/qibotn/backends/quimb.py

```diff
@@ -134,3 +145,143 @@ def execute_circuit(
+            expectation_value = 0.0
+            for ops, sites, coeffs in zip(operators_list, sites_list, coeffs_list):
+                exp_values = circ.local_expectation(
+                    ops, where=sites, backend=self.backend, optimize=self.optimizer
+                )
+                expectation_value += np.dot(coeffs, exp_values)
+            return np.real(expectation_value)
+
+        else:
+            circ_ansatz = qtn.circuit.Circuit
+            circ = circ_ansatz.from_openqasm2_str(circuit.to_qasm())
+            expectation_value = 0.0
+            for ops, sites, coeffs in zip(
+                operators_list_grouped, sites_list_grouped, coeffs_list_grouped
+            ):
+                exp_values = circ.local_expectation(
+                    ops, where=sites, backend=self.backend, optimize=self.optimizer
+                )
+                expectation_value += np.dot(coeffs, exp_values)
+            return np.real(expectation_value)
+
+    @staticmethod
+    def _qiboobs_to_quimbobs(hamiltonian):
+        """
+        Convert a Qibo SymbolicHamiltonian into a Quimb-compatible decomposition.
+
+        Returns three lists:
+        - operators_list: Quimb operators (tensor products of Pauli matrices).
+        - sites_list: tuples of qubit indices the operators act on.
+        - coeffs_list: coefficients for each term.
+        """
+
+        factor_pattern = re.compile(r"([^\d]+)(\d+)")
+
+        operators_list = []
+        sites_list = []
+        coeffs_list = []
+
+        for term in hamiltonian.terms:
+            coeff = term.coefficient
+            term_ops = []
+            term_sites = []
+
+            for factor in term.factors:
+                match = factor_pattern.match(str(factor))
+                if not match:
+                    raise ValueError(
+                        f"Factor '{str(factor)}' does not match the expected format."
+                    )
+
+                operator_name = match.group(1)
+                qubit_index = int(match.group(2))
+
+                # Build the single-qubit operator
+                if operator_name not in {"X", "Y", "Z", "I"}:
+                    raise ValueError(f"Unsupported operator {operator_name}")
+                op = qu.pauli(operator_name)
+
+                term_ops.append(op)
+                term_sites.append(qubit_index)
+
+            # Build the tensor product if more than one factor
+            if term_ops:
+                full_op = term_ops[0]
+                for op in term_ops[1:]:
+                    full_op = full_op & op
+            else:
+                # Identity term (just coefficient)
+                full_op = qu.eye(2)
+
+            operators_list.append(full_op)
+            sites_list.append(tuple(term_sites))
+            coeffs_list.append(coeff)
+
+        return operators_list, sites_list, coeffs_list
+
+    @staticmethod
+    def _group_by_tuples(A, B, C):
```

Comment:
I would make this more clear by renaming 

- `A` --> `qubit_tuples`;
- `B` --> `operator_lists`;
- `C` --> `coeff_lits`.